core/node/eth_watch/src/client.rs

```diff
@@ -173,14 +147,9 @@ impl EthClient for EthHttpQueryClient {
 
                 // safety check to prevent infinite recursion (quite unlikely)
                 if from_number >= mid {
-                    return Err(Error::InfiniteRecursion);
+                    return result;
```

Comment:
If this code gets broken in future, won't having an explicit error help us to find it quickly? Even if so, I'm not sure if we should keep it.

Reply:
I've added a `WARN` log before returning the result. WDYT, is this a good way to deal with the issue?

Reply:
warning should be enough, thanks